addons/hr_holidays/models/resource.py

```diff
@@ -64,14 +64,14 @@ def _reevaluate_leaves(self, time_domain_dict):
         })
         self.env.add_to_compute(self.env['hr.leave']._fields['number_of_days'], leaves)
         self.env.add_to_compute(self.env['hr.leave']._fields['duration_display'], leaves)
-        sick_time_status = self.env.ref('hr_holidays.holiday_status_sl')
+        sick_time_status = self.env.ref('hr_holidays.holiday_status_sl', raise_if_not_found=False)
         for previous_duration, leave, state in zip(previous_durations, leaves, previous_states):
             duration_difference = previous_duration - leave.number_of_days
             message = False
             if duration_difference > 0 and leave.holiday_status_id.requires_allocation == 'yes':
                 message = _("Due to a change in global time offs, you have been granted %s day(s) back.", duration_difference)
             if leave.number_of_days > previous_duration\
-                    and leave.holiday_status_id not in sick_time_status:
+                    and (sick_time_status and leave.holiday_status_id not in sick_time_status):
```

Comment:
This does not work and will never be triggered if the sick time status is deleted; it should be done like this 
```suggestion
                    and (not sick_time_status or leave.holiday_status_id not in sick_time_status):
```

Reply:
> This does not work and will never be triggered if the sick time status is deleted; it should be done like this

The changes have been made according to your suggestions. Thank you :)